### PR TITLE
[18.0][MIG] sale_commercial_partner: Migration to 18.0

### PR DESCRIPTION
**Pull request by BertVGroenendael**

### CODE REVIEW

File: sale_commercial_partner/views/sale_order.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="name">commercial.partner.sale.order.form</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <field name="partner_id" position="before">
+                <field name="commercial_partner_id" invisible="1" />
+            </field>
+        </field>
+    </record>
```
Comment (LucasTran380381) [inline sale_commercial_partner/views/sale_order.xml:11]: recommend to removed ( not need to add invisible field)

**Comment by LucasTran380381** [comment]
Pls squash odoo bot commit

File: sale_commercial_partner/tests/test_sale_commercial_partner.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Copyright (C) 2018 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests import Form, common
+
+
+class TestSaleCommercialPartner(common.TransactionCase):
+    def setUp(self):
```
Comment (StefanRijnhart) [inline sale_commercial_partner/tests/test_sale_commercial_partner.py:8]: Thanks for adding the test! Can you refactor to use @classmethod setUpClass(cls)?

**Review by LucasTran380381 — CHANGES_REQUESTED** [review]

File: sale_commercial_partner/views/sale_order.xml (hunk 1 above)
Comment (StefanRijnhart) [reply sale_commercial_partner/views/sale_order.xml:11]: No, please keep it. We use it in inherits that include the field in domains and such.

Comment (LucasTran380381) [reply sale_commercial_partner/views/sale_order.xml:11]: In odoo18,  domain field will automatically added invisible field that mean no need to declared invisible field to use it in domain

Comment (StefanRijnhart) [reply sale_commercial_partner/views/sale_order.xml:11]: Excellent, thanks for clarifying! I tested this successfully, using the field in `domain` and `invisible` attributes. Found the implementation in https://github.com/odoo/odoo/pull/137031. I'll add it to the OCA's 18.0 migration guidelines.

Just curious, do you know if this is documented anywhere?

Comment (LucasTran380381) [reply sale_commercial_partner/views/sale_order.xml:11]: No documentation, recommend dev using odoo-module-migrator and follow guidelines.

**Comment by rousseldenis** [comment]
@BertVGroenendael Could you attend @StefanRijnhart comment ?

**Comment by LucasTran380381** [comment]
Since this PR has been inactive for 2 weeks, I've started working on related PRs based on this approach
https://github.com/OCA/sale-workflow/pull/3504 @rousseldenis @StefanRijnhart

**Comment by StefanRijnhart** [comment]
Closing in favor of https://github.com/OCA/sale-workflow/pull/3504

**Pull request closed without merging**